### ISSUE TITLE
[3.3] Regression: No .htaccess throws exception when Apache checks are disabled.

### ISSUE THREAD

**Issue by rossriley** [bolt/bolt#6455]
This is a pretty new one, only just showing up after a recent pull of release/3.3

Loading up the frontend now throws the following:
![screen shot 2017-03-04 at 14 12 08](https://cloud.githubusercontent.com/assets/5082/23579415/a4c12542-00e4-11e7-8ea8-bfa9462e4c09.png)

This is still happening when the Apache checks are disabled as detailed in the exception.

**Comment by GawainLynch** [comment]
Yep, sorry this is from #6429 and something that we thought we had "caught" … it is literally what invoked the 5% quantity versus 95% effort phenomenon on that PR.

Quoth me in private conversation during said PR work:

> We get this one wrong, and we'll hear about it from Ross is seconds

@CarsonF assigning this one to you mate :wink:

**Comment by rossriley** [comment]
Ha, well I'm not sure I want to take up this role as ogre...

I think I can spot the problem, the check in the new code is purely on the existence of the Apache server software, whereas I hope there's plenty of us that run Apache but disallow .htaccess use for performance/security reasons.

**Comment by GawainLynch** [comment]
Haha … Not ogre, just that I broken it on the implementation of the validators and you hit it quickly, as you're one of the sensible ones that turns off (and doesn't use) `.htaccess`. `<insert pro-Nginx rant here>`

One of the things I battle with is if we even should have that check at all any longer (before I get jumped on by others, I know some people use web hosting and the `.htaccess` file is their only option).

My thought being that the check dates back to v1 approaches, and everything has changed since then, etc.

**Comment by rossriley** [comment]
Is there any way we can check the behaviour, ie that url rewrites are working as expected.

Obviously there's multiple server and config setups and it does seem silly to check that 1: A user is running Apache and then that they are using an .htaccess file when even the existence of those two things gives no guarantee that Bolt will still work.

Whereas if we could check that url requests are getting redirected to the right place, then it doesn't matter how it's done, we just know that it is.

Obviously it might be slower triggering off a sub-request, but maybe we could do it once and cache in the same way we do the db check etc.

**Comment by CarsonF** [comment]
I'll see if I can figure out why the old way isn't working.

Besides BC, the "new" way, for now, is this:
```php
$app['config.validator.apache_enabled'] = false;
```

**Comment by rossriley** [comment]
OK, I've got to the bottom of this. The crux is that whereas the configuration used to have a separate verifier object depending on the type of install, the new code has the `disableApacheChecks` method on the config object itself, so

We can either:

1: Leave as is and change the instructions in the exception message to:

    you can disable this check
    by calling <code>$config->disableApacheChecks();</code>

that is skipping the `getVerifier` on the `$config` option.

2: If we want to make it completely BC then we can tweak the method `getVerifier` in both `ResourceManager` and the sub-class `Configuration'\Composer` to just return itself rather than passing on to the ComposerChecks or LowlevelChecks object respectively.

That enables the original `$config->getVerifier->disableApacheChecks();` to work as before.

**Comment by CarsonF** [comment]
@rossriley Did you look [here](https://github.com/bolt/bolt/blob/release/3.3/src/Configuration/LowlevelChecks.php#L60-L69)? I don't think that's the problem.

**Comment by CarsonF** [comment]
@rossriley Can you post your bootstrap snippet where the check fails to disable?

**Comment by CarsonF** [comment]
Nevermind I found it. I missed the `disableApacheChecks()` method. I thought you just did `disableApacheChecks = true` since the property was public.

**Comment by rossriley** [comment]
No, it doesn't run through there any more...

This is the line:
https://github.com/bolt/bolt/blob/release/3.3/src/Configuration/ResourceManager.php#L196

And it runs on the config object itself and not on the checks class.

Here's the bootstrap code.

```
$configuration = new Bolt\Configuration\Composer(__DIR__);
$configuration->setPath("web", "public");
$configuration->setPath("files", "public/files");
$configuration->getVerifier()->disableApacheChecks();
$configuration->verify();

$app = new App\BoltApplication(array('resources'=>$configuration));
```

If you just change it to:

```php
$configuration = new Bolt\Configuration\Composer(__DIR__);
$configuration->setPath("web", "public");
$configuration->setPath("files", "public/files");
$configuration->disableApacheChecks();
$configuration->verify();

$app = new App\BoltApplication(array('resources'=>$configuration));
```
Then all works again.

**Comment by GawainLynch** [comment]
Fixed in #6528
